pkg/syncer/cnsoperator/controller/csinodetopology/csinodetopology_controller.go

```diff
@@ -168,19 +203,32 @@ var _ reconcile.Reconciler = &ReconcileCSINodeTopology{}
 type ReconcileCSINodeTopology struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver.
-	client      client.Client
-	scheme      *runtime.Scheme
-	configInfo  *cnsconfig.ConfigurationInfo
-	recorder    record.EventRecorder
-	useNodeUuid bool
+	client              client.Client
+	scheme              *runtime.Scheme
+	configInfo          *cnsconfig.ConfigurationInfo
+	recorder            record.EventRecorder
+	useNodeUuid         bool
+	enableTKGsHAinGuest bool
+	vmOperatorClient    client.Client
+	supervisorNamespace string
 }
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 // Note: The Controller will requeue the Request to be processed again if the
 // returned error is non-nil or Result.Requeue is true, otherwise upon
 // completion it will remove the work from the queue.
-func (r *ReconcileCSINodeTopology) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+func (r *ReconcileCSINodeTopology) Reconcile(ctx context.Context, request reconcile.Request) (
+	reconcile.Result, error) {
+	if r.enableTKGsHAinGuest {
+		return r.reconcileForGuest(ctx, request)
```

Comment:
Do you have different logic for `return r.reconcileForVanilla(ctx, request) ` and `r.reconcileForVanilla(ctx, request)`?

Reply:
They are pretty much the same except controller in vanilla queries VCenter TagManager while controller in guest queries the VMOP in supervisor. 

But, I wrote this way to try not to change any behavior in vanilla.